features/keychain/module/crypto/secp256k1.js

```diff
@@ -1,22 +1,40 @@
 import assert from 'minimalistic-assert'
 import elliptic from '@exodus/elliptic'
 import { mapValues, pick } from '@exodus/basic-utils'
+import ecc from '@exodus/bitcoinerlab-secp256k1'
 
-const validEcOptions = (ecOptions) =>
+import { tweakPrivateKey, tweakPublicKey } from './tweak'
+
+const isValidEcOptions = (ecOptions) =>
   !ecOptions || Object.keys(ecOptions).every((key) => ['canonical'].includes(key))
 
+const isValidTweakOptions = (tweakOptions) =>
+  !tweakOptions ||
+  Object.keys(tweakOptions).every((key) => ['tweakHash', 'extraEntropy'].includes(key))
+
 export const create = ({ getPrivateHDKey }) => {
   const EC = elliptic.ec
   const curve = new EC('secp256k1')
 
   const createInstance = () => ({
     signBuffer: async ({ seedId, keyId, data, ecOptions, enc = 'der' }) => {
       assert(['der', 'raw'].includes(enc), 'signBuffer: invalid enc')
-      assert(validEcOptions(ecOptions), 'signBuffer: invalid EC option')
+      assert(isValidEcOptions(ecOptions), 'signBuffer: invalid EC option')
       const { privateKey } = getPrivateHDKey({ seedId, keyId })
       const signature = curve.sign(data, privateKey, pick(ecOptions, ['canonical']))
       return enc === 'der' ? Buffer.from(signature.toDER()) : { ...signature }
     },
+    signSchnorr: async ({ seedId, keyId, data, tweak = false, tweakOptions }) => {
+      assert(isValidTweakOptions(tweakOptions), 'signSchnorr: invalid tweak options')
+      const hdkey = getPrivateHDKey({ seedId, keyId })
+      const privateKey = tweak ? tweakPrivateKey({ hdkey, tweakOptions }) : hdkey.privateKey
+      return ecc.signSchnorr(data, privateKey, tweakOptions?.extraEntropy)
+    },
+    getPublicKey: async ({ seedId, keyId, tweak = false, tweakOptions }) => {
+      assert(isValidTweakOptions(tweakOptions), 'getPublicKey: invalid tweak options')
+      const hdkey = getPrivateHDKey({ seedId, keyId })
+      return tweak ? tweakPublicKey({ hdkey, tweakOptions }) : hdkey.publicKey
+    },
```

Comment:
Note: Consumers can tweak the public key themselves, we technically don't need to do this on the keychain itself.

https://github.com/ExodusMovement/bitcoinerlab-secp256k1/blob/exodus/index.js#L190-L203

Reply:
@kewde I cannot confirm this. I cannot reconstruct the tweaked public key by only using the normal public key and the tweak hash. I may be doing something wrong.

Reply:
@kewde , I would love it if we can tweak the public key outside the keychain, but right now I can't get it to work. Can we continue with the current implementation?

Reply:
That's fine with me 👍

Reply:
Done: https://github.com/ExodusMovement/exodus-oss/pull/90/commits/2d762c8ea2d0fc7a58a8d56beae31e82834208e0